draft text for project goals, please comment

dvdsk opened the pull request:
reviews might be a good way to discuss this. We can merge it once we largely agree. 

For big changes/additions a PR is probably best.

alice-i-cecile commented on goals.md:
```diff
@@ -0,0 +1,18 @@
+We believe all audio needs can be shared by one base. Currently there are
+multiple audio engines. They all where made with different goals in mind.
+
+# Goals
```
This is more of "why"

alice-i-cecile commented on goals.md:
```diff
@@ -0,0 +1,18 @@
+- combining the best ideas and code from existing projects.
+
+# Definitions
+- performing well: maintaining low latency without audible audio glitches. This
```
This feels like the start of a "what" section?

alice-i-cecile commented on goals.md:
```diff
@@ -0,0 +1,18 @@
+We believe all audio needs can be shared by one base. Currently there are
+multiple audio engines. They all where made with different goals in mind.
```
```suggestion
multiple audio engines. They were all made with different goals in mind.
```

alice-i-cecile commented on goals.md:
```diff
@@ -0,0 +1,19 @@
+- combining the best ideas and code from existing projects.
+
+# What
+- a crate being a good base: We can implement all the needed use cases with
```
What are the core use cases? Who wants to use audio crates in Rust? Are there key tradeoffs in the space?

dvdsk replied:
Yes we should have those here. I have some answers to those questions but they are bound to be incorrect. I propose we 

- collect all possible use cases. If possible with some user story/context around those. To that end I have added what I know from Rodio.
- document the tradeoffs
- select which to focus on (the core use case).

alice-i-cecile reviewed (commented):
Generally could use  a rewrite for grammar / writing quality: ping me when the content is done and I'll be happy to draft something up for you quick.

BillyDM commented:
I think it would be helpful to explain how the Firewheel engine can be used to meet all of these goals. If you don't mind, I would like to write up my own draft text for project goals coming from own perspective. Then we can look into merging our drafts together.

dvdsk commented:
> I think it would be helpful to explain how the Firewheel engine can be used to meet all of these goals. If you don't mind, I would like to write up my own draft text for project goals coming from own perspective. Then we can look into merging our drafts together.

Good idea, I've added you to the repo feel free to push directly to main or open a PR. Let me know if you need anything.

BillyDM commented:
Alright, I added my own text to the document!

dvdsk commented:
I am still working on incorporating the changes. Looking back *project goals* as a title was not a good starting point :) 
I'll let everyone know when its ready for another review :+1:

dvdsk commented:
I've moved things around. There is now one short Readme with multiple supporting pieces.

- I felt like the project goals should be the first you see so it has largely replaced the Readme text. 
- I have also introduced a distinction between requirements and the proposed tech stack. This makes more sense to me as the requirements should lead to the right stack for audio in rust.

We have a good base now, provided there are no major objections I would like to merge this as is. Further changes would then be new new PR's.

alice-i-cecile reviewed (approved):
No major objections. Bother me for an editing pass before widespread publication but I agree on merging.
